lib-js/directory.ts

```diff
@@ -0,0 +1,102 @@
+import yaml from "js-yaml"
+import { promises as fs } from "fs"
+import path from "path"
+
+import { withArchive } from "./hrx"
+
+interface YamlOptions {
+  ":todo"?: string[]
+  ":ignore_for"?: string[]
+  ":warning_todo"?: string[]
+  ":precision"?: number
+}
+
+function hasOptionForImpl(option: string[] | undefined, impl: string) {
+  if (!option || !(option instanceof Array)) return false
+  return option.some((item) => item.includes(impl))
+}
+
+interface OverrideOptions {
+  todoWarning?: boolean
+  mode?: "ignore" | "todo"
+  precision?: number
+}
+
+function getOptionOverrides(options: YamlOptions, impl: string) {
+  const opts: OverrideOptions = {}
+  if (hasOptionForImpl(options[":warning_todo"], impl)) {
+    opts.todoWarning = true
+  }
+  if (hasOptionForImpl(options[":ignore_for"], impl)) {
+    opts.mode = "ignore"
+  }
+  if (hasOptionForImpl(options[":todo"], impl)) {
+    opts.mode = "todo"
+  }
+  if (options[":precision"]) {
+    opts.precision = options[":precision"]
+  }
+  return opts
+}
+
+interface DirOptions {
+  impl: string
+  bin: string
+  rootDir: string
+  testDir: string
+  // TODO guarantee that this is `run` or `probe`
+  todoMode?: string
+}
+
+type AllOptions = DirOptions & OverrideOptions
+
+type DirIteratee = (dir: string, options: AllOptions) => Promise<void>
```

Comment:
Style nit: I'd just define this inline, I think that's clearer than making it a named type if you aren't going to reuse it.